processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -62,14 +67,26 @@
    *
    * @return a new Factory that can be used for operations on top of data output from the current factory.
    */
-  public AggregatorFactory getCombiningFactory();
+  public abstract AggregatorFactory getCombiningFactory();
+
+  /**
+   * Returns an AggregatorFactory that can be used to merge the output of aggregators from this factory and
+   * other factory.
+   * This method is relevant only for AggregatorFactory which can be used at ingestion time.
+   *
+   * @return a new Factory that can be used for merging the output of aggregators from this factory and other.
+   */
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    throw new UnsupportedOperationException("can't merge");
```

Comment:
minor comment, it'd be nice if this printed the type of base agg factory and other agg factory